pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -183,17 +207,22 @@ func (r *reconcilerBase) upsertDeployment(ctx context.Context, reconcilerRef typ
 func (r *reconcilerBase) createOrPatchDeployment(ctx context.Context, declared *appsv1.Deployment) (controllerutil.OperationResult, error) {
 	dRef := client.ObjectKeyFromObject(declared)
 	kind := "Deployment"
-
-	current := &appsv1.Deployment{}
-
-	if err := r.client.Get(ctx, dRef, current); err != nil {
+	forcePatch := true
```

Comment:
What does this force patch actually do? Take over fields that are managed by another manager?

Should we instead just error if another owner has taken control of our fields?

Reply:
@nan-yu do you have perspective on this?

Reply:
https://kubernetes.io/docs/reference/using-api/server-side-apply/#using-server-side-apply-in-a-controller 
It is strongly recommended for controllers to always "force" conflicts, since they might not be able to resolve or act on these conflicts.

Reply:
This means we make sure to own the fields we defined.